### PR TITLE
Backend: performance improvements for large datasets (after profiling umsp)

### PR DESCRIPTION
**Pull request by d-callan**
so far the primary change is to check column types before attempting to update them. ex, if its already a date, dont try to update it to date..

### CODE REVIEW

**Comment by d-callan** [comment]
im considering adding a shadow col for dates. theyd be read in as character types, which is pretty quick. then lubridate::as_date could fairly quickly convert them to Date types, but id leave the character col under another name. then when we go to write to json we dont have the costly conversion back to character. 
probably makes sense to have this an optional arg to updateTypes helper. bc i dont think the histo needs it, but the scatter def does..

**Comment by d-callan** [comment]
still considering how, or if i should, convert the outliers and rawData options for box to data.table syntax as well as the best fit lines for scatter.. 
i probably should.. reporting outliers is now the most costly part of the 3.5s boxplot of 4mil points.. and ive been too scared to profile scatter w smoothed conditional means yet..

**Comment by d-callan** [comment]
hmm. heres a better idea i think, for dates. it takes ~7s to convert 4mil Date to character to meet the api using as.character. but i could conditionally use either as.character (for numerics) or format (for Dates).. format still takes ~1.5s over 4mil points, but it doesnt mean holding 4mil points in memory in duplicate.

**Comment by d-callan** [comment]
and w that last, our 4mil scatter of age v date is taking about ~7.2s locally for me now. that time includes fread and writing the json output file.

**Comment by d-callan** [comment]
so smoothed means are exactly as terrifying as i feared.. currently clocking 137s in R, for a single group of 4mil points.. 124s of that is work done largely in C, and probably not further optimizable.. I can profile w multiple groups and at least make sure we dont add unnecessary overhead that way.

**Review by asizemore — APPROVED** [review]
Had one general question but doesn't need to prevent merge. Looks good to me!

File: R/class-plotdata-box.R [hunk 1]
```diff
@@ -77,11 +77,18 @@ newBoxPD <- function(.dt = data.table::data.table(),
       .pd.base <- cbind(.pd.base, outliers)
     }
   } else if (points == 'all') {
-    aggStr <- getAggStr(y, c(x, group, panel))
-    rawData <- data.table::as.data.table(aggregate(as.formula(aggStr), .pd, list))
+    byCols <- colnames(.pd)[colnames(.pd) %in% c(x, group, panel)]
+    rawData <- .pd[, list(rawData=lapply(.SD, as.vector)), keyby=byCols]
+
+    if (length(byCols)) {
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/class-plotdata-box.R:83]: Hmm. Think youre right, bc while it's technically possible to build a box plot without groups along the x-axis, we've actively disallowed it in both the data service and plot.data validation. So I could change it and we could revisit if we ever change our minds about the x-axis in box. (box margins in subset tab might for ex, idk yet)

**Pull request merged**